### ISSUE TITLE
NZ hierarchy issues

### ISSUE THREAD

**Issue by missinglink** [whosonfirst-data/whosonfirst-data#1921]
Hi @stepps00 ,

There seem to be some issues in New Zealand with how the placetypes are being defined 🤷‍♂️ 
It's common to find the same place twice in WOF appearing in two different placetypes.

Looking in my local area as an example:

- [Wellington City](https://spelunker.whosonfirst.org/id/101914285/) exists on the `locality` level
- [Wellington City](https://spelunker.whosonfirst.org/id/102079339/) also exists on the `county` level.

Looking at the [Suburbs of Wellington City](https://en.wikipedia.org/wiki/Category:Suburbs_of_Wellington_City):
- [Karori](https://spelunker.whosonfirst.org/id/1360019659/) exists as a `locality`
- [Karori](https://spelunker.whosonfirst.org/search/?q=karori&iso=nz&placetype=neighbourhood) again as a `neighbourhood`.
- [Keburn](https://spelunker.whosonfirst.org/id/85772991/) only as a `neighourhood`
- [Makara](https://spelunker.whosonfirst.org/id/101915011/) only as a `locality`.

I think this is maybe why I'm always getting confused as to whether a metro area in NZ/AU should be categorised as:
1. The city is a `county` containing Suburbs which are `locality`
2. The city is a `locality` containing Suburbs which are `neighbourhood`

My understanding after my rambling on https://github.com/whosonfirst-data/whosonfirst-data-admin-au/pull/24#issuecomment-697373030 was that these should be mapped as per 1.

It looks like they have kinda been mapped as both 1. and 2. which has resulted in some duplication.

What do you think, is this easy to fix or do we need to go through and manually check them?

**Comment by missinglink** [comment]
I think actually a lot of our PIP issues where `Wellington` is being assigned as the `locality` for addresses instead of the Suburb will naturally resolve themselves since the Dec 2020 change to make [Wellington City](https://github.com/whosonfirst-data/whosonfirst-data-admin-nz/blob/master/data/101/914/285/101914285.geojson) on the `locality` layer a `Point` geometry.

I'm guessing there is a reason this record continues to exist on the `locality` placetype, either to provide a centroid which cannot be added to the `county` record for some reason or to indicate the "Wellington Central Business District (CBD)" although that would usually be mapped as a Polygon encompassing the downtown retail area?

**Comment by missinglink** [comment]
Here's `Auckland` for comparison:

- [Auckland City](https://spelunker.whosonfirst.org/id/102079403/) `county`
- [Auckland City](https://spelunker.whosonfirst.org/id/101914257/) `locality`

https://en.wikipedia.org/wiki/List_of_suburbs_of_Auckland

- [Ponsonby](https://spelunker.whosonfirst.org/id/85773127/) `neighbourhood`
- [Celevedon](https://spelunker.whosonfirst.org/id/101914783/) `locality`
- Totara Vale - not found
- [Murrays Bay](https://spelunker.whosonfirst.org/id/85778477/) `locality`
- [Mount Roskill](https://spelunker.whosonfirst.org/id/85773043/) `neighbourhood`

It looks like all the central Suburbs were made into a `neighbourhood` and all the outer Suburbs a `locality`, is that correct?

**Comment by stepps00** [comment]
Thanks for the issue @missinglink.

When trying to debug some of these issues, I realized the Spelunker is actually out of date in some cases (though the distribution files should be up-to-date). I think it's related to https://github.com/whosonfirst/whosonfirst-www-spelunker/issues/145.

That being said, a few comments:

>Looking in my local area as an example:
>
>- [Wellington City](https://spelunker.whosonfirst.org/id/101914285/) exists on the `locality` level
>- [Wellington City](https://spelunker.whosonfirst.org/id/102079339/) also exists on the `county` level.

With this Wellington example, we've also introduced some `localadmin` records, Wellington included: 

https://github.com/whosonfirst-data/whosonfirst-data-admin-nz/blob/master/data/172/923/858/3/1729238583.geojson

So we've got county and localadmin polygons, as well as a locality point. This is the case in a handful of other urban areas of New Zealand, too. The county represents the second admin-level (I think these are district-equivalent), the localadmin represents the "core" area (created by dissolving on the `city_id` attribute in our `nz-linz` data), and a locality point (it has a curated boundary box and zoom levels for map display).

>Looking at the [Suburbs of Wellington City](https://en.wikipedia.org/wiki/Category:Suburbs_of_Wellington_City):
>- [Karori](https://spelunker.whosonfirst.org/id/1360019659/) exists as a `locality`
>- [Karori](https://spelunker.whosonfirst.org/search/?q=karori&iso=nz&placetype=neighbourhood) again as a `neighbourhood`.

Karori is now classified as a locality, along with other suburbs like Thomdom, Kelburn, and Te Aro. The [neighbourhood of Karori](https://spelunker.whosonfirst.org/id/85772989/) was flagged as "not current" and superseded into the new locality.

>- [Keburn](https://spelunker.whosonfirst.org/id/85772991/) only as a `neighourhood`

This record was actually deprecated, along with other Quattroshapes-sourced neighbourhood data in NZ.

>- [Makara](https://spelunker.whosonfirst.org/id/101915011/) only as a `locality`.

Yep, this is another suburb we're classifying as a locality.

> I think actually a lot of our PIP issues where Wellington is being assigned as the locality for addresses instead of the Suburb will naturally resolve themselves since the Dec 2020 change to make Wellington City on the locality layer a Point geometry.

That's great! Let me know if the above comments are in line with expectations - the records should all have updated hierarchies. 

> I'm guessing there is a reason this record continues to exist on the locality placetype, either to provide a centroid which cannot be added to the county record for some reason or to indicate the "Wellington Central Business District (CBD)" although that would usually be mapped as a Polygon encompassing the downtown retail area?

Yes, because the locality is a "common" placetype in Who's On First, we kept a current record for these localities as Point records, so services using the data can display the place appropriately on a map without untangling various properties/placetypes. 

> It looks like all the central Suburbs were made into a neighbourhood and all the outer Suburbs a locality, is that correct?

I believe we've generally made the suburbs into localities.. if you're seeing otherwise, I'd double check the `mz:is_funky`, `mz:is_current` and edtf/superseding properties.

To summarize, I don't think there are any issues that need to be fixed through this issue, but there are some records that need to be updated as part of https://github.com/whosonfirst-data/whosonfirst-data/issues/1920. Let me know if you have any other questions though, happy to take a look at some other examples.

**Comment by missinglink** [comment]
Agh you are completely right, I think the confusion is due to the Spelunker being out-of-date.
I spot checked a couple and they seem to be correct on Github so I'm going to close this, sorry for the noise.

**Comment by stepps00** [comment]
No worries, thanks for these issues @missinglink. Let us know if you find any other issues in NZ.
